frontend/src/components/admin/question-creation/question-elements/modals/fraction/FractionModal.tsx

```diff
@@ -2,47 +2,62 @@ import React, { useEffect, useState } from "react";
 import { FormControl, FormErrorMessage, FormLabel } from "@chakra-ui/react";
 
 import type { FractionMetadata } from "../../../../../../types/QuestionMetadataTypes";
+import type { FractionType } from "../../../../../../types/QuestionTypes";
 import { stringToFloat } from "../../../../../../utils/GeneralUtils";
 import FractionInput from "../../../../../common/fraction/FractionInput";
 import Modal from "../../../../../common/modal/Modal";
 
 interface FractionModalProps {
   isOpen: boolean;
   onClose: () => void;
+  onBack?: () => void;
   onConfirm: (data: FractionMetadata) => void;
   data?: FractionMetadata;
+  fractionType: FractionType;
 }
 
 const FractionModal = ({
   isOpen,
   onClose,
+  onBack,
   onConfirm,
   data,
+  fractionType,
 }: FractionModalProps): React.ReactElement => {
+  const [wholeNumber, setWholeNumber] = useState<string>("");
   const [numerator, setNumerator] = useState<string>("");
   const [denominator, setDenominator] = useState<string>("");
   const [error, setError] = useState(false);
 
   useEffect(() => {
+    setWholeNumber(String(data?.wholeNumber ?? ""));
     setNumerator(String(data?.numerator ?? ""));
     setDenominator(String(data?.denominator ?? ""));
   }, [data]);
 
   const handleClose = () => {
+    setWholeNumber(String(data?.wholeNumber ?? ""));
     setNumerator(String(data?.numerator ?? ""));
     setDenominator(String(data?.denominator ?? ""));
     setError(false);
     onClose();
   };
 
   const handleConfirm = () => {
+    const castedWholeNumber =
+      fractionType === "regular" ? null : stringToFloat(wholeNumber);
     const castedNumerator = stringToFloat(numerator);
     const castedDenominator = stringToFloat(denominator);
```

Comment:
can these be decimal values? should we be casting everything to a float? or is it just easier because other number values are stored as floats in the database?

Reply:
good catch! should be ints

Reply:
they get stored as ints regardless - but will fix code for accuracy

Reply:
i guess we should have validation for no decimals then?